### ISSUE TITLE
[Bug]: Qwen30B MoE模型在vLLM 0.11.0rc1上采集的profile信息异常

### ISSUE THREAD

**Issue by a-seeker** [vllm-project/vllm-ascend#4585]
### Your current environment

详细环境信息：
Package                           Version       Editable project location
--------------------------------- ------------- --------------------------
aiofiles                          25.1.0
aiohappyeyeballs                  2.6.1
aiohttp                           3.13.2
aiosignal                         1.4.0
annotated-doc                     0.0.4
annotated-types                   0.7.0
anyio                             4.11.0
astor                             0.8.1
async-timeout                     5.0.1
attrs                             25.4.0
auto-tune                         0.1.0
blake3                            1.0.8
blinker                           1.9.0
cachetools                        6.2.2
cbor2                             5.7.1
certifi                           2025.11.12
cffi                              2.0.0
charset-normalizer                3.4.4
click                             8.3.1
cloudpickle                       3.1.2
cmake                             4.2.0
compressed-tensors                0.11.0
dataflow                          0.0.1
decorator                         5.2.1
depyf                             0.19.0
dill                              0.4.0
diskcache                         5.6.3
distro                            1.9.0
dnspython                         2.8.0
einops                            0.8.1
email-validator                   2.3.0
exceptiongroup                    1.3.1
fastapi                           0.122.0
fastapi-cli                       0.0.16
fastapi-cloud-cli                 0.5.1
fastar                            0.7.0
filelock                          3.20.0
Flask                             3.1.2
frozendict                        2.4.7
frozenlist                        1.8.0
fsspec                            2025.10.0
gguf                              0.17.1
h11                               0.16.0
h2                                4.3.0
hccl                              0.1.0
hccl-parser                       0.1
hf-xet                            1.2.0
hpack                             4.1.0
httpcore                          1.0.9
httptools                         0.7.1
httpx                             0.28.1
huggingface-hub                   0.36.0
Hypercorn                         0.18.0
hyperframe                        6.1.0
idna                              3.11
interegular                       0.3.3
itsdangerous                      2.2.0
Jinja2                            3.1.6
jiter                             0.12.0
jsonschema                        4.25.1
jsonschema-specifications         2025.9.1
lark                              1.2.2
llguidance                        0.7.30
llvmlite                          0.45.1
lm-format-enforcer                0.11.3
markdown-it-py                    4.0.0
MarkupSafe                        3.0.3
mdurl                             0.1.2
mistral_common                    1.8.5
modelscope                        1.32.0
mpmath                            1.3.0
msadvisor                         1.0.0
msgpack                           1.1.2
msgspec                           0.20.0
multidict                         6.7.0
networkx                          3.4.2
ninja                             1.13.0
numba                             0.62.1
numpy                             1.26.4
op-compile-tool                   0.1.0
op-gen                            0.1
op-test-frame                     0.1
opc-tool                          0.1.0
openai                            2.8.1
openai-harmony                    0.0.8
opencv-python-headless            4.11.0.86
outlines_core                     0.2.11
packaging                         25.0
partial-json-parser               0.2.1.1.post7
pillow                            12.0.0
pip                               25.3
priority                          2.0.0
prometheus_client                 0.23.1
prometheus-fastapi-instrumentator 7.1.0
propcache                         0.4.1
protobuf                          6.33.1
psutil                            7.1.3
py-cpuinfo                        9.0.0
pybase64                          1.4.2
pybind11                          3.0.1
pycountry                         24.6.1
pycparser                         2.23
pydantic                          2.12.4
pydantic_core                     2.41.5
pydantic-extra-types              2.10.6
Pygments                          2.19.2
python-dotenv                     1.2.1
python-json-logger                4.0.0
python-multipart                  0.0.20
PyYAML                            6.0.3
pyzmq                             27.1.0
Quart                             0.20.0
referencing                       0.37.0
regex                             2025.11.3
requests                          2.32.5
rich                              14.2.0
rich-toolkit                      0.16.0
rignore                           0.7.6
rpds-py                           0.29.0
safetensors                       0.7.0
schedule-search                   0.0.1
scipy                             1.15.3
sentencepiece                     0.2.1
sentry-sdk                        2.46.0
setproctitle                      1.3.7
setuptools                        80.9.0
setuptools-scm                    9.2.2
shellingham                       1.5.4
sniffio                           1.3.1
soundfile                         0.13.1
soxr                              1.0.0
starlette                         0.50.0
sympy                             1.14.0
taskgroup                         0.2.2
te                                0.4.0
tiktoken                          0.12.0
tokenizers                        0.22.1
tomli                             2.3.0
torch                             2.7.1+cpu
torch_npu                         2.7.1
torchvision                       0.22.1
tqdm                              4.67.1
transformers                      4.57.2
typer                             0.20.0
typing_extensions                 4.15.0
typing-inspection                 0.4.2
urllib3                           2.5.0
uvicorn                           0.38.0
uvloop                            0.22.1
vllm                              0.11.0+empty  /mnt/data3/fyj/vllm
vllm_ascend                       0.11.0rc2     /mnt/data3/fyj/vllm-ascend
watchfiles                        1.1.1
websockets                        15.0.1
Werkzeug                          3.1.3
wheel                             0.45.1
wsproto                           1.3.2
xgrammar                          0.1.25
yarl                              1.22.0

双机16卡，world_size=16, 并行配置为dp16,tp1,ep16

运行程序：
node0:
python examples/offline_data_parallel.py \
                    --model="/mnt/data2/Qwen3-30B-A3B/" \
                    --dp-size=16 \
                    --tp-size=1 \
                    --node-size=2 \
                    --node-rank=0 \
                    --enable-expert-parallel \
                    --master-addr=192.168.0.111 \
                    --master-port=13345 \
                    --enforce-eager
node1:
python examples/offline_data_parallel.py \
                    --model="/mnt/data2/Qwen3-30B-A3B/" \
                    --dp-size=16 \
                    --tp-size=1 \
                    --node-size=2 \
                    --node-rank=1 \
                    --enable-expert-parallel \
                    --master-addr=192.168.0.111 \
                    --master-port=13345 \
                    --enforce-eager

### 🐛 Describe the bug

代码信息(参考链接：https://github.com/vllm-project/vllm-ascend/blob/main/examples/offline_data_parallel.py)：
```python
import contextlib
import gc
import os
from time import sleep

import torch
from vllm import LLM, SamplingParams
from vllm.distributed.parallel_state import (  # noqa E402
    destroy_distributed_environment, destroy_model_parallel)
from vllm.utils import get_open_port

os.environ["VLLM_USE_MODELSCOPE"] = "True"
os.environ["VLLM_WORKER_MULTIPROC_METHOD"] = "spawn"

def parse_args():
    import argparse

    parser = argparse.ArgumentParser(description="Data Parallel Inference")
    parser.add_argument(
        "--model",
        type=str,
        default="ibm-research/PowerMoE-3b",
        help="Model name or path",
    )
    parser.add_argument("--dp-size",
                        type=int,
                        default=2,
                        help="Data parallel size")
    parser.add_argument("--tp-size",
                        type=int,
                        default=1,
                        help="Tensor parallel size")
    parser.add_argument("--node-size",
                        type=int,
                        default=1,
                        help="Total number of nodes")
    parser.add_argument("--node-rank",
                        type=int,
                        default=0,
                        help="Rank of the current node")
    parser.add_argument("--master-addr",
                        type=str,
                        default="",
                        help="Master node IP address")
    parser.add_argument("--master-port",
                        type=int,
                        default=0,
                        help="Master node port")
    parser.add_argument("--enforce-eager",
                        action="store_true",
                        help="Enforce eager mode execution.")
    parser.add_argument("--trust-remote-code",
                        action="store_true",
                        help="Trust remote code.")
    parser.add_argument("--enable-expert-parallel",
                        action="store_true",
                        help="Enable expert parallel, used in MOE models.")
    return parser.parse_args()

def cleanup_env_and_memory():
    destroy_model_parallel()
    destroy_distributed_environment()
    with contextlib.suppress(AssertionError):
        torch.distributed.destroy_process_group()
    gc.collect()
    torch.npu.empty_cache()
    torch.npu.reset_peak_memory_stats()

def main(
    model,
    dp_size,
    local_dp_rank,
    global_dp_rank,
    dp_master_ip,
    dp_master_port,
    GPUs_per_dp_rank,
    enable_expert_parallel,
    enforce_eager,
    trust_remote_code,
):
    # DP only support on V1 engine
    os.environ["VLLM_DP_RANK"] = str(global_dp_rank)
    os.environ["VLLM_DP_RANK_LOCAL"] = str(local_dp_rank)
    os.environ["VLLM_DP_SIZE"] = str(dp_size)
    os.environ["VLLM_DP_MASTER_IP"] = dp_master_ip
    os.environ["VLLM_DP_MASTER_PORT"] = str(dp_master_port)
    os.environ['VLLM_TORCH_PROFILER_DIR'] = '/mnt/data3/fyj/vllm-ascend/profiler/dp4_tp4_ep16/max_token_16/npu_moe_distribute_dispatch_v2'

    prompts = ["The operation $\\times$ is defined for all nonzero numbers by $a \\otimes b = \\frac{a^{2}}{b}$. Determine $[(1 \\otimes 2) \\otimes 3] - [1 \\otimes (2 \\otimes 3)]$"] * 128
    if len(prompts) == 0:
        # if any rank has no prompts to process,
        # we need to set a placeholder prompt
        prompts = ["Placeholder"]
    print(f"DP rank {global_dp_rank} needs to process {len(prompts)} prompts")

    # Create a sampling params object.
    # since we are doing data parallel, every rank can have different
    # sampling params. here we set different max_tokens for different
    # ranks for demonstration.
    sampling_params = SamplingParams(temperature=0.8,
                                     top_p=0.95,
                                     max_tokens=16)

    # Create an LLM.
    llm = LLM(
        model=model,
        tensor_parallel_size=GPUs_per_dp_rank,
        enforce_eager=enforce_eager,
        enable_expert_parallel=enable_expert_parallel,
        trust_remote_code=trust_remote_code,
        tokenizer="/mnt/data2/Qwen3-30B-A3B/",
        tokenizer_mode="slow"
    )

    llm.start_profile()

    outputs = llm.generate(prompts, sampling_params)

    llm.stop_profile()
    # Print the outputs.
    for i, output in enumerate(outputs):
        if i >= 5:
            # print only 5 outputs
            break
        prompt = output.prompt
        generated_text = output.outputs[0].text
        print(f"DP rank {global_dp_rank}, Prompt: {prompt!r}, "
              f"Generated text: {generated_text!r}")

    # Give engines time to pause their processing loops before exiting.
    sleep(5)
    del llm
    cleanup_env_and_memory()

if __name__ == "__main__":
    args = parse_args()

    dp_size = args.dp_size
    tp_size = args.tp_size
    node_size = args.node_size
    node_rank = args.node_rank

    if node_size == 1:
        dp_master_ip = "127.0.0.1"
        dp_master_port = get_open_port()
    else:
        dp_master_ip = args.master_addr
        dp_master_port = args.master_port

    assert dp_size % node_size == 0, "dp_size should be divisible by node_size"
    dp_per_node = dp_size // node_size

    from multiprocessing import Process

    procs = []
    for local_dp_rank, global_dp_rank in enumerate(
            range(node_rank * dp_per_node, (node_rank + 1) * dp_per_node)):
        proc = Process(
            target=main,
            args=(
                args.model,
                dp_size,
                local_dp_rank,
                global_dp_rank,
                dp_master_ip,
                dp_master_port,
                tp_size,
                args.enable_expert_parallel,
                args.enforce_eager,
                args.trust_remote_code,
            ),
        )
        proc.start()
        procs.append(proc)
    exit_code = 0
    for proc in procs:
        proc.join(timeout=900)
        if proc.exitcode is None:
            print(
                f"Killing process {proc.pid} that didn't stop within 15 minutes."
            )
            proc.kill()
            exit_code = 1
        elif proc.exitcode:
            exit_code = proc.exitcode

    exit(exit_code)
```
观测到的主要问题及异常点（意外且耗时的集合通信操作 (Allreduce/Allgather)）：
如下图所示，在时间轴上出现了名为 Hcom_allgather和 Hcom_allreduce的操作，并且这些通信操作在整个时间片内占据了显著的时间。但是我的并行配置信息是world_size 16 dp16 tp1 ep16,并不存在tp，为什么会有allreduce.

<img width="865" height="498" alt="Image" src="https://github.com/user-attachments/assets/736097a8-1ae4-4f2b-9b6f-456805f11404" />
在未开启 TP 的情况下，出现集合通信（如 Allreduce 或 Allgather）是异常的。这些通信操作导致了极长的等待时间 (Notify Wait) 和整体耗时。
通算融合算子执行时间异常：
	算子名称： MoeDistributeDispatchV2 和 MoeDistributeCombineV2。
	MoeDistributeDispatchV2 的 Wall Duration (墙钟时长) 约为17 ms 。
	MoeDistributeCombineV2 的 Wall Duration 约为 4ms
对于这种通算融合算子，17ms 的执行时间是过长的。而在 vLLM 0.9.1 版本下，这些算子的执行时间通常只要在500us左右的。

<img width="865" height="274" alt="Image" src="https://github.com/user-attachments/assets/6e1bcfc4-f9c6-41f2-ba4e-04de5b057ef8" />

<img width="865" height="288" alt="Image" src="https://github.com/user-attachments/assets/489c8dcb-f40d-458a-88c9-01118232d5f5" />
整体推理性能严重下降：
一层 Decode 操作需要62ms。一次完整的 Decode 过程总时长需要2s左右。该推理速度0.5token/s是无法接受的，严重低于预期性能。

<img width="865" height="453" alt="Image" src="https://github.com/user-attachments/assets/048e3a96-4127-4f08-92df-410e113b3c0c" />
基于上述现象，问题的核心猜测可能集中在profile采样工具上。因为我们基于verl框架进行了一轮后训练，时间如下图所示：

<img width="865" height="99" alt="Image" src="https://github.com/user-attachments/assets/54acaa31-8959-4f2c-9b55-c7faa488a2e1" />
Decode长度平均为1000左右，时间在354s,3token/s这个速度是可以接受的。这里说明推理框架本身应该没有问题。

**Comment by Nagisa125** [comment]
Thank you for your feedback. Does this issue still exist in the latest version? Has the problem been resolved?
